runtime-decompiler/src/main/java/org/jrd/frontend/MainFrame/VmDecompilerInformationController.java

```diff
@@ -196,12 +208,95 @@ private void loadClassBytecode(String name) {
         bytecodeDecompilerView.reloadTextField(decompiledClass);
     }
 
+    private static String lastFile = System.getProperty("user.home");
+
+    private void rewriteClass(String name) {
+        try {
+            if (name == null || name.trim().isEmpty()) {
+                name = "???";
+            }
+            final JDialog jd = new JDialog((JFrame) null, "Specify class and select its bytecode", true);
+            jd.setSize(400, 300);
+            jd.setLayout(new BorderLayout());
+            final JPanel inputs = new JPanel(new GridLayout(3, 1));
+            final JPanel buttons = new JPanel(new GridLayout(3, 1));
+            final JLabel validation = new JLabel("???");
+            final JTextField filePath = new JTextField(lastFile);
+            final JTextField className = new JTextField(name);
+            DocumentListener v = new FiletoClassValidator(validation, filePath, className);
+            filePath.getDocument().addDocumentListener(v);
+            className.getDocument().addDocumentListener(v);
+            v.changedUpdate(null);
+            final JButton select = new JButton("...");
+            select.addActionListener(new ActionListener() {
+                @Override
+                public void actionPerformed(ActionEvent e) {
+                    JFileChooser jf = new JFileChooser(filePath.getText());
+                    int returnVal = jf.showOpenDialog(null);
+                    if (returnVal == JFileChooser.APPROVE_OPTION) {
+                        filePath.setText(jf.getSelectedFile().getAbsolutePath());
+                    }
+                }
+            });
+            final JLabel nothing = new JLabel();
+            final JButton ok = new JButton("ok");
+            final boolean[] ook = new boolean[]{false};
```

Comment:
Are you creating boolean array just to get a reference? Why don't you use Boolean?

Reply:
Oh... That doesn't work sorry. Doesn't matter you won't have to use hacks like this when you use MVC.